frontend/packages/core/src/Input/text-field.tsx

```diff
@@ -128,6 +170,8 @@ export interface TextFieldProps
     >,
     Pick<MuiInputProps, "readOnly" | "endAdornment"> {
   onReturn?: () => void;
+  isAutoCompleteable?: boolean;
+  autocompleteCallback?: (v: string) => Promise<any>;
```

Comment:
you can get rid of the boolean here and just use the presence of the callback
```suggestion
  autocompleteCallback?: (v: string) => Promise<any>;
```